### PR TITLE
FOUNDATIONS: Retrieve All Interns

### LINKED ISSUE #51: FOUNDATIONS: Retrieve All Interns

### PR DESCRIPTION
**Pull request by Pwongchaiya**
Closed #51

### CODE REVIEW

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Logic.Add.cs [hunk 1]
```diff
@@ -4,6 +4,11 @@
 // ---------------------------------------------------------------
 
 using System;
+// ---------------------------------------------------------------
```
Comment (glhays) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Logic.Add.cs:7]: Remove

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.cs [hunk 2]
```diff
@@ -56,6 +57,10 @@ public static TheoryData MinutesBeforeOrAfter()
         private static Intern CreateRandomIntern(DateTimeOffset dates) =>
             CreateInternFiller(dates).Create();
 
+        private static IQueryable<Intern> CreateRandomInterns() =>
+            CreateInternFiller(dates: GetRandomDateTime()).Create(GetRandomNumber()).AsQueryable();
+
+
```
Comment (glhays) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.cs:63]: Remove extra line

File: InternTrack.Core.Api.Tests.Unit/DeleteMe.cs [hunk 3]
```diff
@@ -1,14 +0,0 @@
-// ----------------------------------------------------------------------------------
```
Comment (glhays) [inline InternTrack.Core.Api.Tests.Unit/DeleteMe.cs:1]: I beleive DeleteMe file can be removed.

Comment (Pwongchaiya) [reply InternTrack.Core.Api.Tests.Unit/DeleteMe.cs:1]: Did that on merge for Retrieve All

File: InternTrack.Core.Api/Services/Foundations/Interns/IInternService.cs [hunk 4]
```diff
@@ -11,5 +12,6 @@ namespace InternTrack.Core.Api.Services.Foundations.Interns
     public interface IInternService
     {
         ValueTask<Intern> AddInternAsync(Intern intern);
+        IQueryable<Intern> RetrieveAllInternAsync();
```
Comment (glhays) [inline InternTrack.Core.Api/Services/Foundations/Interns/IInternService.cs:15]:
```suggestion
        IQueryable<Intern> RetrieveAllInternAsync();
```IQueryable<Intern> RetrieveAllInternsAsync();

File: InternTrack.Core.Api/Services/Foundations/Interns/InternService.cs [hunk 5]
```diff
@@ -28,5 +29,8 @@ public async ValueTask<Intern> AddInternAsync(Intern intern)
         {
             return await this.storageBroker.InsertInternAsync(intern);
         }
+
+        public IQueryable<Intern> RetrieveAllInternAsync() =>
```
Comment (glhays) [inline InternTrack.Core.Api/Services/Foundations/Interns/InternService.cs:33]: Rename via the IInternService renaming.

**Comment by glhays** [comment]
@Pwongchaiya I remember @Hassan once saying.....
A good engineer should read code twice as much as they write code.
Remember to get up and give yourself a break now and then go back with a fresh perspective on what you have written by reading it again, your young and you need your eyes for years to come.

**Comment by Pwongchaiya** [comment]
Screenshot (136)

**Comment by Pwongchaiya** [comment]
Screenshot (137)

**Pull request closed without merging**